uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar.component.js

```diff
@@ -266,59 +282,123 @@ define([
      */
     function monthStructure (dateMoment) {
       return {
-        index: dateMoment.month(),
+        index: getMonthIndex(dateMoment),
+        month: dateMoment.month(),
         year: dateMoment.year(),
         name: dateMoment.format('MMMM'),
-        shortName: dateMoment.format('MMM')
+        moment: moment().year(dateMoment.year()).month(dateMoment.month())
       };
     }
 
     /**
-     * Reloads the selected months data
+     * Paginates the currently selected month in a specified direction
+     *
+     * @param {String} direction previous|next
+     */
+    function paginateMonth (direction) {
+      var monthAction = direction === 'previous' ? 'subtract' : 'add';
+      var dateFromMonth = vm.selectedMonth.moment[monthAction](1, 'month');
+
+      vm.selectedMonthIndex = getMonthIndex(dateFromMonth);
+
+      setSelectedMonth();
+      setMonthPaginatorsAvailability();
+      refresh('month');
+    }
+
+    /**
+     * Reloads the selected month's data
      *
      * If the source of the refresh is a period change, then
      * it rebuilds the months list as well
      * If the source of the refresh is a change in contacts filters, then
      * it reloads the contacts as well
      *
-     * @param {string} source The source of the refresh (period or contacts change)
+     * @param {String} source The source of the refresh (period or contacts change)
      */
     function refresh (source) {
-      source = _.includes(['contacts', 'period'], source) ? source : 'period';
+      source = _.includes(['contacts', 'period', 'month'], source) ? source : 'period';
 
       $q.resolve()
-        .then(function () {
-          vm.loading.calendar = true;
-        })
-        .then(makeSureMonthsAreNotInjected)
+        .then(makeSureMonthIsNotInjected)
         .then(source === 'period' ? buildPeriodMonthsList : _.noop)
+        .then(source === 'period' ? setFirstPeriodMonth : _.noop)
         .then(source === 'contacts' ? loadContacts : _.noop)
+        .then(source === 'month' ? setMonthPaginatorsAvailability : _.noop)
         .then(function () {
-          injectAndShowMonths((source === 'contacts'));
+          injectAndShowMonth((source === 'contacts'));
         });
     }
 
+    function selectCurrentMonth () {
```

Comment:
Missing documentation.

Reply:
Fixed, also renamed function to avoid confusion with the current month setter.